src/porepy/models/solution_strategy.py

```diff
@@ -693,10 +777,40 @@ def assemble_linear_system(self) -> None:
 
         The linear system is defined by the current state of the model.
 
+        If ``params['apply_schur_complement_reduction']`` is True, the
+        :meth:`schur_complement_primary_variables` and
+        :meth:`schur_complement_primary_equations` are used to perform a Schur
+        complement technique.
+
+        See Also:
+
+            - :meth:`~porepy.numerics.ad.equation_system.EquationSystem.
+              assemble_schur_complement_system`
+            - :meth:`~porepy.numerics.ad.equation_system.EquationSystem.assemble`
+
         """
         t_0 = time.time()
-        self.linear_system = self.equation_system.assemble()
-        logger.debug(f"Assembled linear system in {time.time() - t_0:.2e} seconds.")
+
+        if self._apply_schur_complement_reduction():
+            assert self.schur_complement_primary_variables, (
+                "Primary column block for Schur technique not found."
+            )
+            assert self.schur_complement_primary_equations, (
+                "Primary row block for Schur technique not defined."
+            )
+            self.linear_system = self.equation_system.assemble_schur_complement_system(
+                self.schur_complement_primary_equations,
+                self.schur_complement_primary_variables,
+                inverter=cast(
```

Comment:
Please document this possibility to provide an inverter (and that a reasonable default option is provided).

Reply:
Ammended docstring of method.